### ISSUE TITLE
Pillow version requirement for Wagtail 5.0 not correctly specified

### ISSUE THREAD

**Issue by dkirkham** [wagtail/wagtail#10905]
### Issue Summary

Wagtail 5.0.2, Willow 1.5.3 and Pillow 8.4.0 are not compatible, causing this AttributeError when generating renditions:

```
  File ".../venv/lib/python3.8/site-packages/wagtail/images/shortcuts.py", line 13, in get_rendition_or_not_found
    return image.get_rendition(specs)
  File ".../venv/lib/python3.8/site-packages/wagtail/images/models.py", line 426, in get_rendition
    rendition = self.create_rendition(filter)
  File ".../venv/lib/python3.8/site-packages/wagtail/images/models.py", line 511, in create_rendition
    defaults={"file": self.generate_rendition_file(filter)},
  File ".../venv/lib/python3.8/site-packages/wagtail/images/models.py", line 541, in generate_rendition_file
    generated_image = filter.run(
  File ".../venv/lib/python3.8/site-packages/wagtail/images/models.py", line 708, in run
    willow = willow.resize(transform.size)
  File ".../venv/lib/python3.8/site-packages/willow/plugins/pillow.py", line 74, in resize
    return PillowImage(image.resize(size, _PIL_Image().Resampling.LANCZOS))
  File ".../venv/lib/python3.8/site-packages/PIL/Image.py", line 77, in __getattr__
    raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
AttributeError: module 'PIL.Image' has no attribute 'Resampling'
```

The code in question is at https://github.com/wagtail/Willow/blob/stable/1.5.x/willow/plugins/pillow.py#L74:

```python
        return PillowImage(image.resize(size, _PIL_Image().Resampling.LANCZOS))
```

### Steps to Reproduce

1. Install Wagtail bakery demo according to the instructions, modified as follows:
    - prepare a virtual environment with Wagtail 5.0.2, Willow 1.5.3 and Pillow 8.4.0
    - after the clone apply `git checkout ce1d75784136981ed85653af885ab94436eb8040` to get the latest version compatible with Wagtail 5.0.2
2. Run the server and load the home page. A crash dump should appear with the above-mentioned AttributeError.

### Why is this a bug?

Wagtail 5.0.2 specifies `Pillow>=4.0.0,<10.0.0` and `Willow>=1.5,<1.6`.
Willow 1.5.3 does not specify a Pillow version for installation, but specifies `Pillow>=9.1.0,<11.0.0` for testing
If the Python package management does not automatically or correctly upgrade to the latest allowed releases, one could be left with this apparently valid combination of modules, and a failing project.

The Resampling enums were added in Pillow 9.1.0, so Wagtail should specify a dependency of at least 9.1.0.

### Resolution
The workaround is to specify `Pillow>=9.1.0` in the project `requirements.txt`.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes)

### Technical details

-   Python version: 3.8
-   Django version: 4.1
-   Wagtail version: 5.0.2
-   Browser version: not applicable

**Comment by lb-** [comment]
@zerolab or @RealOrangeOne  - could one of you maybe look into this. Not sure if this will be fixed by the upgrade to Willow planned for Wagtail 5.1.*

https://github.com/wagtail/wagtail/pull/10857

**Comment by RealOrangeOne** [comment]
I've looked a bit further into this, and yes it seems like in Pillow 9.1.0 the `Resampling` enum was added. https://github.com/wagtail/Willow/commit/d6fbc3a2387d00ba0f61b8f407c1cae8eef68283 was added to Willow which broke the supported version of Pillow to only allow 9.1.0+.

I agree that the solution here is to bump the version of Pillow. 9.1.0 was released April 2022, so I don't think it's unreasonable to require that as a minimum going forward.

**Comment by zerolab** [comment]
Just saw this. I concur

**Comment by dkirkham** [comment]
So do you think we should update Wagtail 5.0 (ie. 5.0.3) to require Pillow 9.1.0? I'm happy to prepare a PR.

**Comment by zerolab** [comment]
@dkirkham Yes please. We have another fix going to 5.0.x, so this is timely

**Comment by gasman** [comment]
Fixed in #10911
